src/index.js

```diff
@@ -43,7 +43,9 @@ export const schema = {
 
 export const normalize = (input, schema) => {
   if (!input || typeof input !== 'object') {
-    throw new Error(`Unexpected input given to normalize. Expected type to be "object", found "${typeof input}".`);
+    throw new Error(
+      `Unexpected input given to normalize. Expected type to be "object", found "${input && typeof input}".`
```

Comment:
Maybe derive the type from the schema? `array` if  `Array.isArray(schema) || schema instanceof schema.Array`

Reply:
I don't believe an array would trigger this. typeof myarray === 'object' and all arrays are truthy.

Reply:
null, undefined, numbers, strings, symbols, functions (and thus static classes as well) are the only types that would trigger this. maybe some other primitive type I'm not aware of.

Reply:
I mean derive the type for `Expected type to be "${Array.isArray(schema) || schema instanceof schema.Array ? 'array' : 'object'}"`

Reply:
ahh, gotcha. yes this is a good additional improvement

Reply:
but then shouldn't it fail if you get an array when expecting a plain object? currently this check will pass

Reply:
Should I add a respective test for array vs plain object in the if statement?

Reply:
Seems like array and objects are interchangable due to tests like:

```js
    test('normalizes Objects using their values', () => {
      const userSchema = new schema.Entity('user');
      const users = new schema.Array(userSchema);
      expect(normalize({ foo: { id: 1 }, bar: { id: 2 } }, users)).toMatchSnapshot();
    });
```